tests/rbd/test_rbd_dm_cache.py

```diff
@@ -0,0 +1,157 @@
+    import pdb
+
+    try:
+        pdb.set_trace()
+        device_names.append(rbd_obj.image_map(pool_name, image_name)[:-1])
+        pvcreate(client_node, device_names[0])
+        vgcreate(client_node, vg_name, device_names[0])
+
+        for name, size in lv_sizes.items():
+            lvm_create(client_node, name, vg_name, size)
+
+        command = f"echo 'y' | lvconvert --type cache-pool --poolmetadata {vg_name}/meta_disk {vg_name}/cache_disk"
+        client_node.exec_command(cmd=command, sudo=True)
+
+        if cache_type == "cache":
+            # Configure dm-cache
+            command = f"lvconvert --type cache --cachepool {vg_name}/cache_disk {vg_name}/data_disk"
+            client_node.exec_command(cmd=command, sudo=True)
+
+        else:
+            # Configure dm-write cache
+            command = f"lvchange --activate n {vg_name}/cachedisk"
+            client_node.exec_command(cmd=command, sudo=True)
+
+            command = f"lvconvert --type writecache --cachevol {vg_name}/cachedisk {vg_name}/metadisk"
+            client_node.exec_command(cmd=command, sudo=True)
+
+        run_mkfs(
+            client_node=client_node, device_name="/dev/device_names[0]", type="xfs"
+        )
+        command = "mkdir -p {test_folder_path}"
+        client_node.exec_command(cmd=command, sudo=True)
+
+        command = f"mount /dev/device_names[0] {test_folder_path}"
+        client_node.exec_command(cmd=command, sudo=True)
+
+        run_fio(
+            client_node=client_node,
+            filename=test_folder_path,
+            runtime=config.get("runtime", 30),
+            size=config.get("file_size", "100M"),
+        )
+
+        return 0
+    except Exception as err:
+        log.error(err)
+        return 1
+
+    finally:
+        rbd_obj.clean_up(pools=[pool_name])
+        rbd_obj.image_unmap(device_names[0])
+
+
+def run(ceph_cluster, **kw):
+    """
+    Test to configure dm-cache and dm-writecache for RBD images
+    from LVM side to enhance cache mechanism.
+
+    Args:
+        ceph_cluster: ceph cluster object
+        **kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 for failure.
+    """
+    log.info(
+        "Starting CEPH-83575581 - Verify the usage of ceph RBD images as "
+        "dm-cache and dm-write cache from LVM side to enhance cache mechanism."
+    )
+
+    config = kw.get("config")
+    pool_name = config["rep_pool_config"]["pool"]
+    image_name = config["rep_pool_config"]["image"]
+    rbd = initial_rbd_config(**kw)
+    rbd_obj = rbd.get("rbd_reppool")
+    cache_client = get_node_by_id(ceph_cluster, config["client"])
+
+    # To configure dm-cache for rbd image based disk
+    test_configure_cache(
+        rbd_obj,
+        cache_client,
+        pool_name,
+        image_name,
+        cache_type="cache",
+        **kw,
+    )
+
+    # creating separate image to configure dm-write cache
+    image_name = rbd.random_string() + "_test_image"
```

Comment:
If you are taking image name from test suite, try taking both for dm-cache and dm-write-cache. Otherwise may be keep both of them as random values just to keep it uniform.

Otherwise the image name mentioned in test suite will look confusing, as to which cache is it being used for.

Reply:
done.